hadoop-plugin/src/main/groovy/com/linkedin/gradle/azkaban/AzkabanBlockedFlowStatusTask.groovy

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright 2015 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.linkedin.gradle.azkaban;
+
+
+import com.linkedin.gradle.azkaban.client.AzkabanClient;
+import com.linkedin.gradle.azkaban.client.AzkabanStatus;
+import com.linkedin.gradle.util.AzkabanClientUtil;
+import org.gradle.api.DefaultTask;
+import org.gradle.api.GradleException;
+import org.gradle.api.tasks.TaskAction;
+
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+
+/**
+ * AzkabanBlockedFlowStatusTask fetches the status of the flows when they are completed.*/
+class AzkabanBlockedFlowStatusTask extends AzkabanFlowStatusTask {
+
+  private static final Long POLLING_WAIT_TIME = 20000; // default 20s
+
+  /**
+   * The Gradle task action for getting flowStatus from Azkaban.*/
+  @TaskAction
+  void status() {
+    if (azkProject.azkabanUrl == null) {
+      throw new GradleException("Please set azkaban.url in the .azkabanPlugin.json file in your project directory.");
+    }
+    getBlockedAzkabanFlowStatus(AzkabanHelper.readSession());
+  }
+
+  List<String> getFlowsToRun() {
+    if (project.hasProperty("flow")) {
+      if (project.getProperties().get("flow").toString().isEmpty()) {
+        return null;
+      } else {
+        List<String> sortedFlowSet = new ArrayList(
+            new HashSet<String>(Arrays.asList(project.getProperties().get("flow").toString().split(",+"))));
+        Collections.sort(sortedFlowSet);
+        return sortedFlowSet;
+      }
+    } else if (flowsToExecute != null && !flowsToExecute.isEmpty()) {
+      return flowsToExecute;
+    } else {
+      return new ArrayList<String>();
+    }
+  }
+
+/**
+ * getBlockedAzkabanFlowStatus waits for the flow to finish. It polls
+ * the Azkabban server and outputs the job that finished immediately.
+ * @param sessionId
+ */
+  void getBlockedAzkabanFlowStatus(String sessionId) {
+
+    // list of all the flows defined in the project
+    List<String> flows = getSortedFlows(sessionId);
+
+
+    // list of all the recent execution ids of the flows defined in the project
+    List<String> execIds = getRecentFlowExecutionIds(sessionId, flows);
```

Comment:
I will take this up as a separate PR. I'm planning to do some refactoring in the previous couple of PRs that I've sent. I'll refactor it as part of those changes.